src/locale.js

```diff
@@ -123,19 +167,45 @@ export default class Locale {
 
 	/**
 	 * Base for the {@link #t} method.
-	 *
 	 * @private
 	 */
-	_t( str, values ) {
-		let translatedString = translate( this.uiLanguage, str );
+	_t( msgid = '', values = [] ) {
+		return translate( this.uiLanguage, msgid )
+			.replace( /%(\d+)/g, ( match, idx ) => values[ idx ] );
+	}
+
+	/**
+	 * Base for the {@link #ct} method.
+	 * @private
+	 */
+	_ct( msgctxt = '', msgid = '', values = [] ) {
+		const msg = translate( this.uiLanguage, `${ msgid } [context: ${ msgctxt }]` );
+		const regex = /%(\d+)/g;
+		const parts = [];
+
+		while ( true ) {
+			const prevIndex = regex.lastIndex;
+			const match = regex.exec( msg );
+			if ( match == null ) {
+				parts.push( msg.substring( prevIndex ) );
+				break;
+			}
+			parts.push( msg.substring( prevIndex, match.index ) );
+			parts.push( values[ match[ 1 ] ] );
+		}
 
-		if ( values ) {
-			translatedString = translatedString.replace( /%(\d+)/g, ( match, index ) => {
-				return ( index < values.length ) ? values[ index ] : match;
-			} );
+		if ( values.some( value => typeof value == 'object' ) ) {
```

Comment:
I could argue that Boolean type can be casted to a string, however maybe we could also check if a value is a function? Maybe there would be uses for this functionallity.